### PR TITLE
Fuzzy finder: Add help link 

### PR DESCRIPTION
**Pull request by romainfrancois**
### Intent

addresses #11139

### Approach

Add a link to the related article: 

image

### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

**Comment by romainfrancois** [comment]
Perhaps there's a way to only use the cue text that would be evocative enough. We're currently not using it, i.e. we land with: 

image

**Comment by romainfrancois** [comment]
Adding a `?` button 

image

going to https://support.rstudio.com/hc/en-us/articles/200710523-Navigating-Code-in-the-RStudio-IDE

File: src/gwt/src/org/rstudio/studio/client/workbench/codesearch/ui/CodeSearchDialog.java [hunk 1]
```diff
@@ -46,13 +50,18 @@ public CodeSearchDialog(Provider<CodeSearch> pCodeSearch)
...
+      VerticalPanel searchPanel = new VerticalPanel();
+      searchPanel.addStyleName(
          CodeSearchResources.INSTANCE.styles().codeSearchDialogMainWidget());
       codeSearch_ = pCodeSearch_.get();
       codeSearch_.setObserver(this);
-      mainPanel.add(codeSearch_.getSearchWidget());
+      searchPanel.add(codeSearch_.getSearchWidget());
+
+      mainPanel.add(searchPanel);
+      mainPanel.add(new HelpButton(new URL("https://support.rstudio.com/hc/en-us/articles/200710523-Navigating-Code-in-the-RStudio-IDE"), "Fuzzy finder help"));
```
Comment (romainfrancois) [inline src/gwt/src/org/rstudio/studio/client/workbench/codesearch/ui/CodeSearchDialog.java:64]: I don't know how the mapping between rstudioLinkName and the url is done, e.g. 

`new HelpButton("pdf_root_document", constants_.rootDocumentChooserTitle())` -> https://support.rstudio.com/hc/en-us/articles/200486298

Comment (kevinushey) [reply src/gwt/src/org/rstudio/studio/client/workbench/codesearch/ui/CodeSearchDialog.java:64]: We maintain a set of redirects at https://github.com/rstudio/rstudio-web/blob/master/web/rstudio.org/site/links/.htaccess; you could add a redirect to this support article here.

**Comment by kevinushey** [comment]
Could we also see how the UI looks with a Help link on its own line, before the input entry box? E.g.

```
Go to File / Function
--------------------

> [ Text box ]

[?] Using the File / Function Finder
```

As an aside, I'm not sure if we want to use the term "fuzzy finder", just because that's more of an internal / colloquial name.

That said, we do use this sort of Help treatment for some cases where UI space is constrained; e.g. for snippets:

Screen Shot 2022-05-18 at 10 41 55 AM

But where space permits, we usually have a help link with separate text; e.g. `[?] Managing Packages` here:

Screen Shot 2022-05-18 at 10 42 24 AM

**Comment by romainfrancois** [comment]
I like it: 

image

- Maybe the font should be smaller ?
- Probably should be a separate article ?

File: src/gwt/src/org/rstudio/studio/client/workbench/codesearch/ui/CodeSearchDialog.java [hunk 2]
```diff
@@ -46,13 +49,17 @@ public CodeSearchDialog(Provider<CodeSearch> pCodeSearch)
...
-   {     
+   {  
       VerticalPanel mainPanel = new VerticalPanel();
       mainPanel.addStyleName(
          CodeSearchResources.INSTANCE.styles().codeSearchDialogMainWidget());
       codeSearch_ = pCodeSearch_.get();
       codeSearch_.setObserver(this);
       mainPanel.add(codeSearch_.getSearchWidget());
+
+      HelpLink help = new HelpLink(constants_.usingFuzzyFinder(), "navigating_code");
```
Comment (romainfrancois) [inline src/gwt/src/org/rstudio/studio/client/workbench/codesearch/ui/CodeSearchDialog.java:60]: Should this wait until https://github.com/rstudio/rstudio-web/pull/33 is merged ? and keep the actual link in the meantime ?

Comment (kevinushey) [reply src/gwt/src/org/rstudio/studio/client/workbench/codesearch/ui/CodeSearchDialog.java:60]: Just merged the redirect now, thanks!

**Comment by kevinushey** [comment]
> Maybe the font should be smaller ?

I agree the font could be a bit smaller, and if possible the text pushed down a bit more (closer to the bottom of the message box)

What's your opinion on this treatment versus the single `( ? )` button on the right side?

**Comment by romainfrancois** [comment]
It's a bit more explicit on its own line, but it makes the ui less minimalist, which is one thing I really like about fff. I'm overall not convinced the `?` is needed at this point (for SW), esp to advertise a feature that was somehow controversial (https://github.com/rstudio/rstudio/pull/11102#issuecomment-1116029326)

**Comment by kevinushey** [comment]
Gotcha, that makes sense. It's probably reasonable to table this PR for now, and re-consider for later?

**Comment by gtritchie** [comment]
It appears we have decided to not do this for Spotted Wakerobin (the comment suggests that, and the related issue was moved to next milestone). So I'm closing this PR.

**Pull request closed without merging**